src/internal.c

```diff
@@ -140,6 +140,7 @@
 
 
 #define ERROR_OUT(err, eLabel) { ret = (int)(err); goto eLabel; }
+#define CHECK_RET(exp, eLabel) do { if ((ret = (exp))) goto eLabel; } while(0)
```

Comment:
Don't like that ERROR_OUT implicitly takes 'ret' but that boat has sailed.
Prefer that CHECK_RET take 'ret' and 'eLabel' as parameters.
And I prefer that you explicitly test for != 0 too.